sveltekit-authentication/index.md

```diff
@@ -0,0 +1,449 @@
+import { createDirectus, rest  } from "@directus/sdk"; // [!code --]
+import { createDirectus, rest, authentication  } from "@directus/sdk"; // [!code ++]
+import { PUBLIC_APIURL } from '$env/static/public'; // [!code --]
+import { PUBLIC_APIURL,PUBLIC_COOKIE_DOMAIN } from '$env/static/public'; // [!code ++]
+
+function getDirectusInstance(fetch,token) {
+
+  const options = fetch ? { globals: { fetch } } : {};
+
+  const directus = createDirectus(PUBLIC_APIURL, options)
+  .with(authentication('cookie', { credentials: 'include' })) // [!code ++]
+  .with(rest());
+
+  if(token) directus.setToken(token); // [!code ++]
+
+  return directus;
+}
+export default getDirectusInstance;
+
+export const constructCookieOpts = (age) => { // [!code ++]
+	return { // [!code ++]
+			'domain': PUBLIC_COOKIE_DOMAIN, // [!code ++]
+			// send cookie for every page // [!code ++]
+			'path': '/', // [!code ++]
+			// server side only cookie so you can't use `document.cookie` // [!code ++]
+			'httpOnly': true, // [!code ++]
+			// only requests from same site can send cookies // [!code ++]
+			// https://developer.mozilla.org/en-US/docs/Glossary/CSRF // [!code ++]
+			'sameSite': "strict", // [!code ++]
+			// only sent over HTTPS in production // [!code ++]
+			'secure': process.env.NODE_ENV === 'production', // [!code ++]
+			// set cookie to expire after a given time // [!code ++]
+			'maxAge': age // [!code ++]
+		} // [!code ++]
+	} // [!code ++]
+```
+
+Additionally we need to define the Domain the Cookie is valid for in the  `.env` file:
+
+```
+PUBLIC_APIURL= "https://directus.example.com"
+PUBLIC_COOKIE_DOMAIN= "example.com" // [!code ++]
+```
+
+## Setup Login/Signup Form
+
+Let's start the user journey from adding a login/signup form:
+
+```svelte [/signin/+page.svelte]
+<script lang="ts">
+	/** @type {import('./$types').PageData} */
+	import { page } from '$app/stores';
+  let email,password;
+	const redirectedFrom = $page.url.searchParams.get('redirectedFrom')
+</script>
+
+<form
+	action="?/login{redirectedFrom?`&redirectedFrom=${redirectedFrom}`:''}"
+	method="POST">
+	<div>
+		<label for="email">Email</label>
+		<input id="email" name="email" type="text" bind:value={email}	
+		/>
+	</div>
+
+	<div>
+		<label for="password">Password</label>
+		<input id="password" name="password" type="password" required bind:value={password}	
+		/>
+	</div>
+
+	<button type="submit">Log in</button>
+	<button formaction="?/register{redirectedFrom?`&redirectedFrom=${redirectedFrom}`:''}">Register</button>
+</form>
+```
+
+The html template is straight forward. We only have a small function to let us redirect the user to a page he was trying to access before needing to login/signup.
+
+The JavaScript form action below will handle the actual request.
+The Directus SDK is not saving HttpOnly Cookies, thus the Directus API is directly accessed from SvelteKit in order to be able to save the tokens in a secure cookie via SvelteKit's cookie handler.
```

Comment:
```suggestion
The template contains small function to let us redirect the user to a page they were trying to access before needing to login/signup. The JavaScript form action below will handle the request.
```

> The Directus SDK is not saving HttpOnly Cookies, thus the Directus API is directly accessed from SvelteKit in order to be able to save the tokens in a secure cookie via SvelteKit's cookie handler.


Can you rewrite this? Doesn't make sense to me as I read it.

Reply:
"The SvelteKit form action below will handle the actual request.
The Directus SDK is not saving Cookies with the HTTP-only Option enabled and we want to set the Cookie on the server, thus we need to access the Directus API directly from SvelteKit ourself and save the tokens in a secure cookie via SvelteKit's cookie handler."

better? i don't see a way around using SvelteKits cookie handler. sveltekit is in control here what is returned from the server, so the directus SDK is of no help here